libparsec/crates/client/tests/unit/client/update_user_profile.rs

```diff
@@ -0,0 +1,224 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::sync::Arc;
+
+use libparsec_client_connection::test_register_send_hook;
+use libparsec_protocol::authenticated_cmds;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use crate::{Client, ClientUserUpdateProfileError};
+
+use super::utils::client_factory;
+
+async fn update_profile_and_check(profile: UserProfile, user_id: UserID, client: &Arc<Client>) {
+    client.update_user_profile(user_id, profile).await.unwrap();
+
+    let users = client.list_users(false, None, None).await.unwrap();
+    let bob = users.iter().find(|user| user.id == user_id).unwrap();
+    p_assert_eq!(bob.current_profile, profile);
+}
+
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn ok(env: &TestbedEnv) {
```

Comment:
According to [this answer](https://github.com/Scille/parsec-cloud/pull/9418#discussion_r1928457769), Alice should not be able to downgrade its own profile because certificate cannot be self-signed. 

We should probably test that an error is returned for that case.